src/binarySearchTree.js

```diff
@@ -0,0 +1,81 @@
+export default class BinarySearchTree{
+  constructor(){
+    this.head = null
+  }
+
+
+
+  insert(data){
+    if(this.head === null){
+      this.head = new Node(data)
+      return
+    }
+
+    this.locateInsert(data, this.head)
+
+    console.log('insert')
+    console.log('head' + this.head)
+  }
+
+  locateInsert(data, node){
+    if(data > node.data && !!node.right){
+      this.locateSearch(data, node.right)
+    }
+    else if(data <= node.data && !!node.left){
+      this.locateSearch(data, node.left)
+    }
+    if(data > node.data){
+      node.setRight(data)
+    }
+    if(data <= node.data){
+      node.setLeft(data)
+    }
+  }
+
+  search(data){
+    console.log('search')
+    return this.locateSearch(data, this.head)
+  }
+
+  locateSearch(data, node){
```

Comment:
I think this could all be contained in one function 'search'. All that 'search' is currently doing is calling locateSearch.